.python-version

```diff
@@ -0,0 +1 @@
+3.9
```

Comment:
Shouldn’t this decision be left to each developer’s local environment? Supporting multiple Python versions is inconvenient when one is hard-coded in the Git repository.

Reply:
For a package that wants to support multiple Pythons, pinning the oldest supported Python helps enforce the greatest common divisor in terms of language and API level. It’s just a guardrail for the developer so they can’t inadvertently use newer language features, which would break the library for users with older Pythons.

It also makes sure that every developer gets the same linting and typechecking findings.

With that being said, the file is only a hint for tools at venv creation time. If a developer absolutely must override this (even though I think this is a bad idea), they’re free to ignore this file and just create a custom venv with a Python version of their choosing.